src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -0,0 +1,21 @@
+package com.scottlogic;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class AuthorPostSorter implements PostSorter {
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder sortOrder) {
+        if (inputList == null) {
+            return null;
+        } else if (inputList.isEmpty()) {
+            return inputList;
+        }
+        Collections.sort(inputList, Comparator.comparing(UserPost::getAuthor));
```

Comment:
Might be worth being carful with this approach, by adjusting the `inputList` itself you no longer have a safe function as the inputs are changed when it is run. This is generally a bad idea because if you start finding issues and this is the cause they are incredibly hard to debug